### PR TITLE
[CI] run cmocka as part of the static-check-unit-test stage

### PR DESCRIPTION
**Pull request by v1v**
### What

Run UTs for the C in the CI . It doesn't run for the `alpine` yet

### Issues

Caused by https://github.com/elastic/apm-agent-php/issues/264
Blocked by https://github.com/elastic/apm-agent-php/pull/266

### Test

```bash
$ make -f .ci/Makefile prepare build static-check-unit-test
...
$ ls -l build 
total 48
-rw-r--r--  1 vmartinez  staff  4893  5 Nov 14:31 7.2.34-tests-unit-tests-junit.xml
-rw-r--r--  1 vmartinez  staff  5127  5 Nov 14:31 7.2.34-tests-unit-tests-junit.xml.bck
-rw-r--r--  1 vmartinez  staff  6322  5 Nov 14:30 junit.xml

$ head -n 8 build/7.2.34-tests-unit-tests-junit.xml
<?xml version="1.0" encoding="UTF-8" ?>
<testsuites>
  <testsuite name="tests" time="0.000" tests="3" failures="0" errors="0" skipped="0" >
    <testcase name="basic_macros_tests.c: variadic_args_count" time="0.000" >
    </testcase>
    <testcase name="basic_macros_tests.c: if_va_args_empty_else" time="0.000" >
    </testcase>
    <testcase name="basic_macros_tests.c: printf_format_args" time="0.000" >
``` 

### Failures

- There are two test failures when running in the alpine container

```
<testsuites>
  <testsuite name="tests" time="0.000" tests="3" failures="2" errors="0" skipped="0" >
    <testcase name="platform_tests.c: stream_errno" time="0.000" >
      <failure><![CDATA[Assertion that a string contains another one (ignoring case) has failed. haystack: `No error information'. needle: `Unknown'.
platform_tests.c:47: error: Failure!]]></failure>
    </testcase>
    <testcase name="platform_tests.c: stream_errno_overflow" time="0.000" >
    </testcase>
    <testcase name="platform_tests.c: capturing_stack_trace" time="0.000" >
      <failure><![CDATA[0 == 0
/app/src/ext/unit_tests/platform_tests.c:81: error: Failure!]]></failure>
    </testcase>
  </testsuite>
</testsuites>
```

You can easily reproduce it locally with the below command

```
$ DOCKERFILE=Dockerfile.alpine make -f .ci/Makefile unit-test
```

### CODE REVIEW

File: .ci/Jenkinsfile [hunk 1]
```diff
@@ -113,6 +113,23 @@ pipeline {
                   }
                 }
               }
+              stage('PHPC Unit Tests') {
+                when {
+                  expression { return env.DOCKERFILE == 'Dockerfile' }
+                }
```
Comment (v1v) [inline .ci/Jenkinsfile:119]: UTs in alpine failed

```
<testsuites>
  <testsuite name="tests" time="0.000" tests="3" failures="2" errors="0" skipped="0" >
    <testcase name="platform_tests.c: stream_errno" time="0.000" >
      <failure><![CDATA[Assertion that a string contains another one (ignoring case) has failed. haystack: `No error information'. needle: `Unknown'.
platform_tests.c:47: error: Failure!]]></failure>
    </testcase>
    <testcase name="platform_tests.c: stream_errno_overflow" time="0.000" >
    </testcase>
    <testcase name="platform_tests.c: capturing_stack_trace" time="0.000" >
      <failure><![CDATA[0 == 0
/app/src/ext/unit_tests/platform_tests.c:81: error: Failure!]]></failure>
    </testcase>
  </testsuite>
</testsuites>
```

**Review by mdelapenya — APPROVED** [review]

**Review by kuisathaverat — APPROVED** [review]

**Review by cachedout — APPROVED** [review]

File: .ci/Makefile [hunk 2]
```diff
@@ -66,6 +66,15 @@ composer: prepare  ## Run composer
...
 		sh -c '/app/.ci/composer.sh'
 
+.PHONY: unit-test
+unit-test: prepare  ## Run unit-test
+	docker run -t --rm \
+		-u $(CURRENT_UID):$(CURRENT_GID) \
+		-v $(PWD):/app \
+		-w /app \
+		$(IMAGE):${PHP_VERSION}$(SUFFIX) \
+		sh -c '/app/.ci/unit-test.sh'
```
Comment (v1v) [inline .ci/Makefile:76]: Run as much as possible within the docker container and nothing locally

File: .ci/unit-test.sh [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+make
+set +e
+make test
+## Save errorlevel to be reported later on
+ret=$?
+
+## Manipulate JUnit report without multiple testsuites entries.
+for file in /app/build/*-unit-tests-junit.xml; do
+    sed -i.bck ':begin;$!N;s#</testsuites>\n<testsuites>##;tbegin;P;D' "${file}"
+done
```
Comment (v1v) [inline .ci/unit-test.sh:18]: It requires to fix the multiple testsuites entries, that's the reason for saving the errorlevel and running the test-errors files manipulation. See https://gitlab.com/cmocka/cmocka/-/issues/5

**Review by SergeyKleyman — APPROVED** [review]

**Pull request merged**